### PR TITLE
[MM-50007] WorkTemplate: disable "private" button when a template contains a playbook without an enterprise license

### PR DESCRIPTION
**Pull request by JulienTant**
#### Summary
Block the "Private" button if the template contains a playbook and the server license is not E20 or Enterprise.

Also, the design has this selector showing the buttons side-by-side (they were on top of each other so far), so I changed the UI to match what is on figma.

##### QA script

- On a new server with no license
- Open work templates
- Go to the last page of the first template
- the private button is disabled and has a tooltip explaining that you need an enterprise license
- Upload an enterprise license (or start a trial)
- Open work templates
- Go to the last page of the first template
- the private button is available.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-50007

#### Related Pull Requests
image

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: components/work_templates/index.tsx [hunk 1]
```diff
@@ -292,6 +292,7 @@ const WorkTemplateModal = () => {
                     visibility={selectedVisibility}
                     onNameChanged={handleOnNameChanged}
                     onVisibilityChanged={handleOnVisibilityChanged}
+                    template={selectedTemplate}
```
Comment (mhd-sln) [inline components/work_templates/index.tsx:295]: not a strong opinion on this but maybe instead of selectedTemplate we could send templateHasPlaybooks.

**Review by mhd-sln — APPROVED** [review]
Thanks @JulienTant!

File: components/work_templates/index.tsx (hunk 1 above)
Comment (JulienTant) [reply components/work_templates/index.tsx:295]: I actually did consider doing this, but ended up feeling like we would have to calculate the content of the work template to get this info every time, even when we don't need it

**Comment by JulienTant** [comment]
@pvev if you had feedback please ping me on Mattermost, I need that in for Monday :P

**Comment by pvev** [comment]
> @pvev if you had feedback please ping me on Mattermost, I need that in for Monday :P

LGTM! :shipit:

**Pull request merged**